auth/cache/pat.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/absmach/supermq/auth"
+	"github.com/absmach/supermq/pkg/errors"
+	repoerr "github.com/absmach/supermq/pkg/errors/repository"
+	"github.com/redis/go-redis/v9"
+)
+
+const Separator = "_"
+
+type patCache struct {
+	client   *redis.Client
+	duration time.Duration
+}
+
+func NewPatsCache(client *redis.Client, duration time.Duration) auth.Cache {
+	return &patCache{
+		client:   client,
+		duration: duration,
+	}
+}
+
+func (pc *patCache) Save(ctx context.Context, scopes []auth.Scope) error {
+	for _, sc := range scopes {
+		key := generateKey(sc.PatID, sc.OptionalDomainID, sc.EntityType, sc.Operation, sc.EntityID)
+		if err := pc.client.Set(ctx, key, true, pc.duration).Err(); err != nil {
+			return errors.Wrap(repoerr.ErrCreateEntity, err)
+		}
+	}
+
+	return nil
+}
+
+func (pc *patCache) CheckScope(patID, optionalDomainID string, entityType auth.EntityType, operation auth.Operation, entityID string) bool {
+	ctx := context.Background()
```

Comment:
Please add context function 


```suggestion
func (pc *patCache) CheckScope(ctx context.Context, patID, optionalDomainID string, entityType auth.EntityType, operation auth.Operation, entityID string) bool {
```

And use that ctx in redis client